app.coffee

```diff
@@ -16,6 +16,14 @@ if Settings.sentry?.dsn?
 
 sessionRedisClient = redis.createClient(Settings.redis.websessions)
 
+if Settings.processLifeLimitSeconds?
+	getRandomArbitrary = (min, max)->
+		Math.random() * (max - min) + min
+	restartOffsetSeconds = Settings.processLifeLimitSeconds + (Settings.processLifeLimitSeconds * getRandomArbitrary(1, 1.2))
+	PROCESS_TERMINATION_TIME = Date.now() + (restartOffsetSeconds * 1000)
+	console.log("process will start to terminate at #{PROCESS_TERMINATION_TIME}")
```

Comment:
N.B. `Settings.processLifeLimitSeconds` is a string not a number.  This is what I came up with locally while testing it:

- convert to number, use rounding to avoid fractional timestamp
- set a minimum life time to avoid configuration problems
- log human readable date
- increase spread from 1-1.2 to 1-1.5 to reduce overlapping restarts 

```coffee
if Settings.processLifeLimitSeconds?
	getRandomArbitrary = (min, max)->
		Math.random() * (max - min) + min
	processLifeLimitSeconds = parseInt(Settings.processLifeLimitSeconds, 10)
	if processLifeLimitSeconds >= 3600
		restartOffsetSeconds = Math.round(processLifeLimitSeconds * getRandomArbitrary(1, 1.5))
		PROCESS_TERMINATION_TIME = Date.now() + restartOffsetSeconds * 1000
		logger.log "process will start to terminate at #{new Date(PROCESS_TERMINATION_TIME)} after #{restartOffsetSeconds}s"
	else
		logger.error {processLifeLimitSeconds}, "refusing to set a process lifetime less than 1 hour"
```